pkg/controller/dnszone/dnszone_controller_test.go

```diff
@@ -253,7 +253,7 @@ func TestReconcileDNSProviderForGCP(t *testing.T) {
 		},
 		{
 			name:    "Delete managed zone",
-			dnsZone: validDNSZoneBeingDeleted(),
+			dnsZone: validGCPDNSZoneBeingDeleted(),
```

Comment:
We should change all of the dns zones in this test function to use GCP dns zones instead of AWS ones. I am fine with creating a card to do it as a separate PR.

It may also be worth switching to use the builder in pkg/test/dnszone/dnszone.go to build these DNSZones.